azurerm/resource_arm_data_factory_linked_service_sql_server.go

```diff
@@ -0,0 +1,357 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/datafactory/mgmt/2018-06-01/datafactory"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmDataFactoryLinkedServiceSQLServer() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmDataFactoryLinkedServiceSQLServerCreateOrUpdate,
+		Read:   resourceArmDataFactoryLinkedServiceSQLServerRead,
+		Update: resourceArmDataFactoryLinkedServiceSQLServerCreateOrUpdate,
+		Delete: resourceArmDataFactoryLinkedServiceSQLServerDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAzureRMDataFactoryLinkedServiceName,
+			},
+
+			"data_factory_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile(`^[A-Za-z0-9]+(?:-[A-Za-z0-9]+)*$`),
+					`Invalid name for Data Factory, see https://docs.microsoft.com/en-us/azure/data-factory/naming-rules`,
+				),
+			},
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"connection_string": {
+				Type:             schema.TypeString,
+				Required:         true,
+				DiffSuppressFunc: azureRmDataFactoryLinkedServiceConnectionStringDiff,
+				ValidateFunc:     validate.NoEmptyStrings,
+			},
+
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"integration_runtime_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"parameters": {
+				Type:     schema.TypeMap,
+				Optional: true,
+			},
+
+			"annotations": {
+				Type:     schema.TypeList,
```

Comment:
would this make more sense as a Set than a List?

Reply:
I didn't have any issues with ordering and annotations. What're you thinking the other benefit of swapping this would be?

Reply:
then that's fine - some of the Azure API's are picky about ordering and some aren't, it depends on the API unfortunately 🙃